ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -781,41 +834,34 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
                   View Preview Table
               </Clickable>
             </div>
-            {previewDataLoading && <div style={styles.warningMessage}>
-              <Spinner style={{position: 'relative', top: '2rem'}} />
-              <div style={{top: '3rem', position: 'relative'}}>
-                It may take up to few minutes to load the data
-              </div>
-            </div>}
-            {previewList.length > 0 &&
+            {fp.toPairs(previewList).length > 0 &&
```

Comment:
nit: is it necessary to convert to `toPairs` to check the size? Would a `!previewList.isEmpty()` guarantee the non emptiness of `toPairs`?

Reply:
`isEmpty()` appears to not exist on either `Map` or `Array` for some reason, going off the MDN documentation. I could do something else, like maybe `previewList.entries().length > 0` but it is functionally equivalent.